File: 2017_07_01/DrawingBoard画板/DrawingBoard/TestVC1/TestVC1.m
//
//  TestVC1.m
//  DrawingBoard
//
//  Created by xiaoyulong on 17/3/25.
//  Copyright © 2017年 xiaoyulong. All rights reserved.
//

#import "TestVC1.h"
#import "SmoothLineView.h"

@interface TestVC1 ()

@end

@implementation TestVC1

- (void)viewDidLoad {
    [super viewDidLoad];
    // Do any additional setup after loading the view.
    
    SmoothLineView *smoothLineView = [[SmoothLineView alloc] initWithFrame:self.view.bounds];
    [self.view addSubview:smoothLineView];

}

- (void)didReceiveMemoryWarning {
    [super didReceiveMemoryWarning];
    // Dispose of any resources that can be recreated.
}

/*
#pragma mark - Navigation

// In a storyboard-based application, you will often want to do a little preparation before navigation
- (void)prepareForSegue:(UIStoryboardSegue *)segue sender:(id)sender {
    // Get the new view controller using [segue destinationViewController].
    // Pass the selected object to the new view controller.
}
*/

@end

File: 2017_06_01/FBRetainCycleDetectorLearn2/FBRetainCycleDetectorLearn2/TestVC1/TestVC1.swift
//
//  TestVC1.swift
//  FBRetainCycleDetectorLearn2
//
//  Created by xiaoyulong on 2017/4/25.
//  Copyright © 2017年 xiaoyulong. All rights reserved.
//

import UIKit



class TestVC1: UIViewController {

    override func viewDidLoad() {
        super.viewDidLoad()

        // Do any additional setup after loading the view.
        
        let detailVC = DetailVC1()
        
        
        detailVC.completionCallBack = { [weak detailVC] in
            
            detailVC?.nameStr = "xiaoyu"
            
            print("detailVC.nameStr==\(String(describing: detailVC?.nameStr))")
            
            print(self)
        }
        
        navigationController?.pushViewController(detailVC, animated: true)
        
        
        let detector = FBRetainCycleDetector.init()
        detector.addCandidate(self)

        
        

//        print("detector.findRetainCycles()==\(detector.findRetainCycles())")
        
//        print("retainCycles==\(retainCycles)")
        
        
//        FBRetainCycleDetector *detector = [[FBRetainCycleDetector alloc] init];
//        [detector addCandidate:john];
//        NSSet *retainCycles = [detector findRetainCycles];
//        NSLog(@"retainCycles==%@", retainCycles);

        
    }

    override func didReceiveMemoryWarning() {
        super.didReceiveMemoryWarning()
        // Dispose of any resources that can be recreated.
    }
    

    /*
    // MARK: - Navigation

    // In a storyboard-based application, you will often want to do a little preparation before navigation
    override func prepare(for segue: UIStoryboardSegue, sender: Any?) {
        // Get the new view controller using segue.destinationViewController.
        // Pass the selected object to the new view controller.
    }
    */

}
